api-spec/v3/transactions-api.yaml

```diff
@@ -0,0 +1,718 @@
+openapi: 3.0.0
```

Comment:
what about take into account the POST transaction body detail `feature` to be included into this v3 api version? I'm talking about the fact to introduce a detail section into POST transaction to distinguish data for an NPG and a Redirect transaction so that we can remove the hardcoded "orderId" into policies.
It can be just prepared here and v2 policies can call the service v3 version with the "npg detail" so that in the future we can take into account also the detail diversification without introducing a v4 version (and a version misalignment between infra version and service one)

Reply:
yes, this could be a solution. My only concern was about creating a mismatch between apim exposed api and internal api, a situation for which apim v3 POST transaction calls the internal v4 and so on making version matching a bit difficult messy

Reply:
I agree with the comment, this v3 could be the opportunity to fix the orderId field. 
if we were to create a new external v3 to the app, should we create a v4?

Reply:
Renamed to `v2.1`, `v3` addressed in https://github.com/pagopa/pagopa-ecommerce-transactions-service/pull/537